webui/react/src/e2e/models/pages/WorkspaceDetails.ts

```diff
@@ -17,10 +17,17 @@ export class WorkspaceDetails extends BasePage {
    * Visits the project details page.
    * @param {string} [workspaceID] - The Workspace to visit. Defaults to '1' for uncategorized
    */
-  async gotoWorkspace(workspaceID: string = '1', tab: string = '', args = {}): Promise<this> {
+  async gotoWorkspace(workspaceID: number = 1, tab: string = '', args = {}): Promise<this> {
     return await this.goto({ ...args, url: `workspaces/${workspaceID}/${tab}` });
   }
 
+  async getIdFromUrl(): Promise<number> {
+    await this._page.waitForURL(this.url);
+    const matches = new URL(this._page.url()).pathname.match(/(\d+)/);
```

Comment:
Minor thing, I might try to get the right path segment here first, then try to cast to a number. I could see things getting weird if some string was included that someone set to something like "01234"

Reply:
this capture group will match the first digits, but i'll make it more specific.